src/main/java/fi/otavanopisto/kuntaapi/server/index/AbstractIndexHander.java

```diff
@@ -0,0 +1,155 @@
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.transport.client.PreBuiltTransportClient;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import fi.otavanopisto.kuntaapi.server.integrations.KuntaApiConsts;
+import fi.otavanopisto.kuntaapi.server.settings.SystemSettingController;
+
+public abstract class AbstractIndexHander {
+
+  private static final String DEFAULT_INDEX = "kunta-api";
+  private static final String DEFAULT_CLUSTERNAME = "elasticsearch";
+  private static final String[] DEFAULT_HOSTS = new String[] {
+    "localhost:9300"
+  };
+
+  @Inject
+  private SystemSettingController systemSettingController;
+  
+  @Inject
+  private Logger logger;
+
+  private String index;
+  private TransportClient client;
+  
+  @PostConstruct
+  public void init() {
+    String[] hosts = systemSettingController.getSettingValues(KuntaApiConsts.SYSTEM_SETTING_ELASTIC_SEARCH_HOSTS, DEFAULT_HOSTS);
+    String clusterName = systemSettingController.getSettingValue(KuntaApiConsts.SYSTEM_SETTING_ELASTIC_CLUSTER_NAME, DEFAULT_CLUSTERNAME);
+    index = systemSettingController.getSettingValue(KuntaApiConsts.SYSTEM_SETTING_ELASTIC_INDEX, DEFAULT_INDEX);
+    client = createClient(hosts, clusterName);
+  }
+  
+  @PreDestroy
+  public void deinit() {
+    closeClient(client);
+  }
+  
+  protected TransportClient getClient() {
+    return client;
+  }
+  
+  protected byte[] serialize(Indexable indexable) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    try {
+      return objectMapper.writeValueAsBytes(indexable);
+    } catch (JsonProcessingException e) {
+      logger.log(Level.SEVERE, "Failed to serialize indexable object", e);
+    }
+    
+    return new byte[0];
+  }
+  
+  protected String getIndex() {
+    return index;
+  }
+  
+  private TransportClient createClient(String[] hosts, String clusterName) {
+    TransportClient transportClient;
+    
+    System.out.println("Cluster: " + clusterName);
+    
+    Settings settings = Settings.builder()
+      .put("cluster.name", clusterName)
+      .build();
+    
+    transportClient = new PreBuiltTransportClient(settings);
+    
+    for (String host : hosts) {
+      System.out.println("Adding host: " + host);
+      
+      String[] parts = StringUtils.split(host, ':');
+      if (parts.length != 2 || !NumberUtils.isNumber(parts[1])) {
+        logger.severe(String.format("Invalid elastic search host %s, dropped", host));
+      }
+      
+      String name = parts[0];
+      Integer port = NumberUtils.createInteger(parts[1]);
+      transportClient.addTransportAddress(resolveTransportAddress(name, port));
+
+      System.out.println("Added host: " + host);
+    }
+
+    System.out.println("Preparing index");
+    prepareIndex(transportClient);
+    System.out.println("Index prepared");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS106)